### PR TITLE
refactor(rust): tweak `finalized_expr_for_symbol_ref`

### PR DESCRIPTION
**Pull request by hyf0**

### CODE REVIEW

**Comment by hyf0** [comment]
* **#2449** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2449?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2448** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2448?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2447** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2447?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2446** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2446?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @hyf0 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/rolldown/src/module_finalizers/scope_hoisting/mod.rs [hunk 1]
```diff
@@ -86,6 +86,11 @@ impl<'me, 'ast> ScopeHoistingFinalizer<'me, 'ast> {
     symbol_ref: SymbolRef,
     preserve_this_semantic_if_needed: bool,
   ) -> ast::Expression<'ast> {
+    if !symbol_ref.is_declared_in_root_scope(self.ctx.symbol_db) {
+      // No fancy things on scoped symbols.
```
Comment (IWANABETHATGUY) [inline crates/rolldown/src/module_finalizers/scope_hoisting/mod.rs]: root scope symbol also has scope, the `scoped symbol` sounds a little ambiguous, maybe `No fancy things on none root scope symbols` ?

Comment (hyf0) [reply crates/rolldown/src/module_finalizers/scope_hoisting/mod.rs]: Done

**Review by IWANABETHATGUY — APPROVED** [review]

**Comment by hyf0** [comment]
### Merge activity

* **Oct 17, 4:15 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/rolldown/rolldown/2449) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

**Pull request merged**